### PR TITLE
Fix segfault when clocking a FF from a ring oscillator

### PR DESCRIPTION
**Pull request by meiniKi**
Fixes https://github.com/YosysHQ/nextpnr/issues/1158

### CODE REVIEW

File: common/kernel/timing.cc [hunk 1]
```diff
@@ -296,8 +296,8 @@ void TimingAnalyser::identify_related_domains()
 
     // For each clock net identify all nets that can possibly drive it. Compute
     // cumulative delays to each of them.
-    std::function<void(const NetInfo *, dict<IdString, delay_t> &, delay_t)> find_net_drivers =
-            [&](const NetInfo *ni, dict<IdString, delay_t> &drivers, delay_t delay_acc) {
+    std::function<void(const NetInfo *, std::set<IdString> &, dict<IdString, delay_t> &, delay_t)> find_net_drivers =
```
Comment (Ravenslofty) [inline common/kernel/timing.cc:299]:
```suggestion
    std::function<void(const NetInfo *, pool<IdString> &, dict<IdString, delay_t> &, delay_t)> find_net_drivers =
```

File: common/kernel/timing.cc [hunk 2]
```diff
@@ -296,8 +296,8 @@ void TimingAnalyser::identify_related_domains()
...
+            [&](const NetInfo *ni, std::set<IdString> &net_trace, dict<IdString, delay_t> &drivers, delay_t delay_acc) {
```
Comment (Ravenslofty) [inline common/kernel/timing.cc:300]:
```suggestion
            [&](const NetInfo *ni, pool<IdString> &net_trace, dict<IdString, delay_t> &drivers, delay_t delay_acc) {
```

File: common/kernel/timing.cc [hunk 3]
```diff
@@ -371,7 +378,8 @@ void TimingAnalyser::identify_related_domains()
             continue;
 
         dict<IdString, delay_t> drivers;
-        find_net_drivers(ni, drivers, 0);
+        std::set<IdString> net_trace;
```
Comment (Ravenslofty) [inline common/kernel/timing.cc:381]:
```suggestion
        pool<IdString> net_trace;
```

**Review by Ravenslofty — COMMENTED** [review]
nextpnr tends to avoid `std::set` because it can introduce nondeterminism across platforms. Does it matter in this case? Perhaps not, but better safe than sorry.

**Comment by meiniKi** [comment]
> nextpnr tends to avoid `std::set` because it can introduce nondeterminism across platforms. Does it matter in this case? Perhaps not, but better safe than sorry.

Thanks for the hint.

**Pull request merged**